src/Avalonia.Controls/Window.cs

```diff
@@ -321,6 +332,26 @@ void SetIsEnabled(IEnumerable<Window> windows, bool isEnabled)
             }
         }
 
+        void SetWindowStartupLocation()
+        {
+            if (WindowStartupLocation == WindowStartupLocation.CenterScreen)
+            {
+                if (PlatformImpl != null)
+                {
+                    var positionAsSize = PlatformImpl.MaxClientSize / 2 - ClientSize / 2;
```

Comment:
Maybe add a comment here saying "This should be using a Screen API, but we don't have one yet and MaxClientSize is the best we have".